packages/gestalt/src/Checkbox.js

```diff
@@ -69,6 +71,14 @@ export default class Checkbox extends React.Component<Props, State> {
     this.props.onChange({ event, checked });
   };
 
+  handleClick = (event: SyntheticInputEvent<>) => {
+    const { onClick } = this.props;
+    if (onClick) {
+      const { checked } = event.target;
```

Comment:
i think this might be better with `event.currentTarget` since the polymorphic typing on the event type is applied to that instead of `event.target`